### DISCUSSION TITLE
Bug after upgrade ruTorrent 4.3.4 (probably > 4.x) on logoff plugin - RESOLVED - It is the best idea ?

### DISCUSSION THREAD

**Discussion by DavidFirefox** [Novik/ruTorrent#2676] [Q&A] [answered]
Hello,
Thanks for your great work and plugins 🥇, and for all the contributors. I use it for long years ago !

I'm just upgrade my 3.x to 4.3.4, and I have a bug, I think it started with ruTorrent > 4.x.
I not a programmer, I have logoff 1.3.

I have resolve the issue and post it on logoff page : https://github.com/exrat/logoff/issues/1

But I'm not sure, it is the best way to do this : 
 I have 2 issues in init.php : 
- line 2 : `eval(getPluginConf("logoff"));`
Error in Apache log : ` AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function getPluginConf() in <directory>/plugins/logoff/init.php:2\nStack trace:\n#0 <directory>/php/getplugins.php(521): require_once()\n#1 {main}\n  thrown in <directory>/plugins/logoff/init.php on line 2',
`
I replaced it by `eval(FileUtil::getPluginConf($plugin["name"]));`
I'm not sure if name is the correct value ? but it work and $plugin["logoff"]) don't work

- line 4 : `$me = getUser();`
Error in Apache log : `AH01071: Got error 'PHP message: PHP Fatal error:  Uncaught Error: Call to undefined function getUser() in <directory>/plugins/logoff/init.php:4\nStack trace:\n#0 <directory>/php/getplugins.php(521): require_once()\n#1 {main}\n  thrown in <directory>/plugins/logoff/init.php on line 4'
`
I replaced by `$me = $_SERVER['REMOTE_USER'];`
It work well, and the logout popup show my login name !

Is it the best patch ?

Thanks in advance
David

**Comment by stickz** [comment, marked as answer]
Hello @DavidFirefox. We simply moved these functions to the `FileUtil` and `User` classes in version 4. You can reference them accordingly. We did this to avoid loading functions we're not using. This has performance benefits for `ruTorrent v4`.

The fix for line 2 should be `eval(FileUtil::getPluginConf("logoff"));`
The fix for line 4 should be `$me = User::getUser();`

**Comment by DavidFirefox** [comment]
Thanks a lot, it work without warning (my patch have one warning).
But when we click on disconnect we don't see the name of the actual user, but it seem to be the same with my config.
I think I'm mixed up this popup with an other, sorry.

Have a good day.

**Comment by DavidFirefox** [comment]
I make an uptodate fork : https://github.com/DavidFirefox/logoff
